tests/sites_tests/tests.py

```diff
@@ -86,6 +86,43 @@ def test_get_current_site_no_site_id(self):
         site = get_current_site(request)
         self.assertEqual(site.name, "example.com")
 
+    @override_settings(SITE_ID='', ALLOWED_HOSTS=['example.com', 'example.net'])
+    def test_get_current_site_no_site_id_and_handle_port_fallback(self):
+        request = HttpRequest()
+
+        s1 = self.site
+        s2 = Site.objects.create(domain='example.com:80', name='example.com:80')
+
+        # Test for host header without port:
+        request.META = {'HTTP_HOST': 'example.com'}
+        site = get_current_site(request)
+        self.assertIsInstance(site, Site)
```

Comment:
They were to distinguish between `Site` and `RequestSite` because `get_current_site()` can return either dependent on whether `django.contrib.sites` is within `INSTALLED_APPS`. I will remove them as other tests already cover this and it doesn't contribute anything extra here.